### DISCUSSION TITLE
Expand alias support in CryptoCreate & CryptoTransfer Transactions

### DISCUSSION THREAD

**Discussion by mgarbs** [hiero-ledger/hiero-improvement-proposals#593] [Edits]
Opening up the discussion around Expand alias support in CryptoCreate & CryptoTransfer Transactions.

From @Nana-EC :

Abstract

Through the implementation of [HIP 32](https://hips.hedera.com/hip/hip-32), the Hedera Network supports the auto creation of accounts using an ECDSA or ED25519 public key to establish an account alias. This alias can be used as a unique identifier for a Hedera account as part of crypto transfers before and after it is created in place of the `shard.realm.num` accountId format.

Notably, today the Hedera network supports 4 forms of account identifiers

1. `Hedera account id` - `shard.realm.num` format. This is the main account identifier for the network across the ecosystem.
2. `Hedera account long-form` - This is the bytes of the cryptographic public key of the account, appended to the shard and realm in the form `shard.realm.bytes`. Currently, this can only be specified via the `auto-create` flow using the `CryptoTransfer` transaction. 
3. `Hedera account long-zero address` - This is the Hedera account id styled as an Ethereum account hex string. The Hedera account id numbers are hex encoded and the value is prefixed with zeros to form a 20 byte address. This was added to represent all accounts in an `EVM` conformant manner.
4. `Ethereum account address` / `public-address` - This is the rightmost 20 bytes of the 32 byte `Keccak-256` hash of the `ECDSA` public key of the account. This calculation is in the manner described by the Ethereum Yellow Paper. Note, that the recovery id is not formally part of the public key and not included in the hash. This is calculated on the consensus nodes using the `ECDSA` key provided in the auto create `CryptoTransfer` transaction.

**Comment by tonimateos** [comment]
We fully support this HIP, it would solve problems we will encounter when onboarding users from other EVM-compatible chains, and avoid putting in place non-standard workflows for them (such as figuring out how to derive the public key from web3 wallets that only show public addresses). Looking forward to its implementation. We'll keep track.
